requirements_tools/upgrade_requirements.py

```diff
@@ -93,19 +99,34 @@ def cleanup_dir(dirname):
 
 def make_virtualenv(args):
     with cleanup_dir(tempfile.mkdtemp()) as tempdir:
-        venv, python, pip = dirs(tempdir)
+        venv, python, pip, pip_tool = venv_paths(tempdir, args.pip_tool)
+        pip_tool = tuple(shlex.split(pip_tool))
+
         print_call(
             sys.executable, '-m', 'virtualenv', venv,
             '-p', args.python, '--never-download',
         )
 
-        def pip_install(*argv):
-            print_call(pip, 'install', '-i', args.index_url, *argv)
-
         # Latest pip installs python3.5 wheels
-        pip_install('pip', 'setuptools', '--upgrade')
-        pip_install('-r', 'requirements-minimal.txt')
-        pip_install('-r', 'requirements-dev-minimal.txt')
+        print_call(
+            pip, 'install',
+            '-i', args.index_url,
+            'setuptools', args.install_deps, '--upgrade'
```

Comment:
this should probably always upgrade `pip` as well

Reply:
`pip install pip pip` seems to be ok actually, not sure how correct it is, or if that will change in the future